### PR TITLE
AP_OSD: Add height above terrain panel and warning

### PR DESCRIPTION
**Pull request by Hwurzburg**
User requested this...Alex is working on adding it to the Frsky telem also

### CODE REVIEW

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 1]
```diff
@@ -1851,6 +1869,19 @@ void AP_OSD_Screen::draw_vtx_power(uint8_t x, uint8_t y)
     }
     backend->write(x, y, false, "%4hu%c", powr, SYM_MW);
 }
+#if AP_TERRAIN_AVAILABLE
+void AP_OSD_Screen::draw_hgt_abvterr(uint8_t x, uint8_t y)
+{
+    AP_Terrain &terrain = AP::terrain();
+
+    float terrain_altitude;
+    if (terrain.height_above_terrain(terrain_altitude,true)) {
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD_Screen.cpp:1878]: would you mind fixing the indentation?

File: libraries/AP_OSD/AP_OSD.h [hunk 2]
```diff
@@ -215,9 +219,7 @@ class AP_OSD_Screen : public AP_OSD_AbstractScreen
     void draw_aspd1(uint8_t x, uint8_t y);
     void draw_aspd2(uint8_t x, uint8_t y);
     void draw_vspeed(uint8_t x, uint8_t y);
-#if HAL_PLUSCODE_ENABLE
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD.h:218]: why is this changing things related to pluscode?

**Review by tridge — CHANGES_REQUESTED** [review]
very nice feature! I only found minor issues. 
can be merged when fixed and tested

File: libraries/AP_OSD/AP_OSD.h (hunk 2 above)
Comment (Hwurzburg) [reply libraries/AP_OSD/AP_OSD.h:218]: historical....havent the foggiest....do I have to figure this one out?

File: libraries/AP_OSD/AP_OSD_Screen.cpp (hunk 1 above)
Comment (Hwurzburg) [reply libraries/AP_OSD/AP_OSD_Screen.cpp:1878]: will run astyle and fix everything...

File: libraries/AP_OSD/AP_OSD.cpp [hunk 3]
```diff
@@ -169,8 +170,16 @@ const AP_Param::GroupInfo AP_OSD::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("_BTN_DELAY", 20, AP_OSD, button_delay_ms, 300),
 #endif
-
+#if AP_TERRAIN_AVAILABLE
+    // @Param: _W_TERR
+    // @DisplayName: Terrain warn level
+    // @Description: Set level below which HGT_TER item will flash. -1 disables
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD.cpp:176]: need units in meters

File: libraries/AP_OSD/AP_OSD.h (hunk 2 above)
Comment (Hwurzburg) [reply libraries/AP_OSD/AP_OSD.h:218]: will fix

File: libraries/AP_OSD/AP_OSD_Screen.cpp (hunk 1 above)
Comment (Hwurzburg) [reply libraries/AP_OSD/AP_OSD_Screen.cpp:1878]: yes

**Review by Hwurzburg — COMMENTED** [review]
@peterbarker think its ready to merge now

File: libraries/AP_OSD/AP_OSD.h (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_OSD/AP_OSD.h:218]: This still seems to be playing around with PLUSCODE stuff?

File: libraries/AP_OSD/AP_OSD.h [hunk 4]
```diff
@@ -218,7 +220,6 @@ class AP_OSD_Screen : public AP_OSD_AbstractScreen
 #if HAL_PLUSCODE_ENABLE
     void draw_pluscode(uint8_t x, uint8_t y);
 #endif
-
```
Comment (peterbarker) [inline libraries/AP_OSD/AP_OSD.h:221]: Random whitespace change

File: libraries/AP_OSD/AP_OSD.h [hunk 5]
```diff
@@ -250,6 +251,8 @@ class AP_OSD_Screen : public AP_OSD_AbstractScreen
     void draw_callsign(uint8_t x, uint8_t y);
     void draw_current2(uint8_t x, uint8_t y);
     void draw_vtx_power(uint8_t x, uint8_t y);
+    void draw_hgt_abvterr(uint8_t x, uint8_t y);
+
```
Comment (peterbarker) [inline libraries/AP_OSD/AP_OSD.h:255]: Random whitespace addition

**Review by peterbarker — APPROVED** [review]
Other trailing whitespace was removed - which is OK by me so long as we never accept patches that *add* it.

But let's try to avoid random whitespace changes; it clouds the git history.
